x/stakeibc/keeper/msg_server_submit_tx.go

```diff
@@ -358,9 +359,28 @@ func (k Keeper) GetLightClientTimeSafely(ctx sdk.Context, connectionID string) (
 }
 
 // Submits an ICQ to get a validator's exchange rate
-func (k Keeper) QueryValidatorExchangeRate(ctx sdk.Context, chainId, validatorAddress string) error {
+func (k Keeper) QueryValidatorExchangeRate(
+	ctx sdk.Context,
+	chainId string,
+	validatorAddress string,
+	callbackDataBz []byte,
+	aggressiveTimeout bool,
+) error {
 	k.Logger(ctx).Info(utils.LogWithHostZone(chainId, "Submitting ICQ for validator exchange rate to %s", validatorAddress))
 
+	// If this query is executed from LSM, there should be a more aggressive timeout since it's UX blocking,
+	// and, in the event of a timeout, we should still enter the callback so we can alert the user that the query failed
+	// If this query is not for an LSM liquid stake, we can have a more relaxed timeout
+	var timeoutDuration time.Duration
+	var timeoutPolicy icqtypes.TimeoutPolicy
+	if aggressiveTimeout {
+		timeoutDuration = LSMSlashQueryTimeout
+		timeoutPolicy = icqtypes.TimeoutPolicy_EXECUTE_QUERY_CALLBACK
```

Comment:
If I were invoking `QueryValidatorExchangeRate`, I would be surprised to learn that setting `aggressiveTimeout` determines the timeout policy

Reply:
do you think [this](https://github.com/Stride-Labs/stride/pull/837#discussion_r1245645683) would be better

Reply:
Lgtm